src/Utilities/api.js

```diff
@@ -142,14 +142,20 @@ export const fetchSystemHistory = (inventoryId, limit) => {
 export const fetchExecutiveReport = async () => {
     await window.insights.chrome.auth.getUser();
 
-    const url = new URL(`${ROS_API_ROOT}${EXECUTIVE_REPORT_API}`,  window.location.origin);
+    const url = new URL(EXECUTIVE_REPORT_API,  window.location.origin);
 
-    return fetch(url).then((res) => {
-        if (!res.ok) {
-            throw Error(res.statusText);
-        }
+    return fetch(url, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json'
+        },
 
-        return res;
-    }).then(res =>  res.json());
+        body: JSON.stringify({
+            service: 'ros',
+            template: 'executiveReport'
+        })
+    })
+    .then(handleErrors)
+    .then((response) => response.blob());
```

Comment:
The promise handling is good enough as it is. An extra then branch does not do anything.

Reply:
@r14chandra since we are using async/await in the `generateExecutivePDFReport` method this extra step is not needed. The `await fetchExecutiveReport()` will resolve the response to blob. you can [read more about it here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#description). Hope this helps!